pkg/api/pluginproxy/access_token_provider_test.go

```diff
@@ -139,6 +147,9 @@ func TestAccessToken(t *testing.T) {
 
 		var authCalls int
 		apiHandler.HandleFunc("/oauth/token", func(w http.ResponseWriter, req *http.Request) {
+			Convey("Should render header template", t, func() {
+				So(req.Header.Get("x-header"), ShouldEqual, "my test header")
+			})
```

Comment:
Not a big fan of having a convey inside of the handler/function. Would suggest assigning the request headers to a variable that you later can use to verify. Similar manner as the authCalls being used.

Reply:
Agree with @marefr.